mutiny-core/src/labels.rs

```diff
@@ -0,0 +1,492 @@
+use crate::error::MutinyError;
+use crate::indexed_db::MutinyStorage;
+use crate::nodemanager::NodeManager;
+use bitcoin::{Address, XOnlyPublicKey};
+use lnurl::lightning_address::LightningAddress;
+use lnurl::lnurl::LnUrl;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use uuid::Uuid;
+
+const LABELS_MAP_KEY: &str = "labels";
+const LABEL_PREFIX: &str = "label/";
+const CONTACT_PREFIX: &str = "contact/";
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, Ord, PartialEq, PartialOrd, Hash)]
+pub struct LabelItem {
+    pub addresses: Vec<Address>,
```

Comment:
Yeah I only did on-chain for this PR because that will be the hardest. With invoices we don't have to worry about labels transfering from a previous invoice to the next so figured that can be a followup.